superset-frontend/src/dashboard/components/nativeFilters/FilterBar/index.tsx

```diff
@@ -352,166 +233,56 @@ const FilterBar: React.FC<FiltersBarProps> = ({
-                        t(
-                          'Click the button above to add a filter to the dashboard',
-                        )
-                      }
-                    />
-                  </FilterBarEmptyStateContainer>
-                ) : (
-                  <FilterControls
-                    dataMaskSelected={dataMaskSelected}
-                    directPathToChild={directPathToChild}
-                    onFilterSelectionChange={handleFilterSelectionChange}
-                  />
-                )}
-              </AntdTabs.TabPane>
-              <AntdTabs.TabPane
-                disabled={!!editFilterSetId}
-                tab={t('Filter sets (%(filterSetCount)d)', {
-                  filterSetCount: filterSetFilterValues.length,
-                })}
-                key={TabIds.FilterSets}
-                css={tabPaneStyle}
-              >
-                <FilterSets
-                  onEditFilterSet={setEditFilterSetId}
-                  disabled={!isApplyDisabled}
-                  dataMaskSelected={dataMaskSelected}
-                  tab={tab}
-                  onFilterSelectionChange={handleFilterSelectionChange}
-                />
-              </AntdTabs.TabPane>
-            </StyledTabs>
-          ) : (
-            <div css={tabPaneStyle} onScroll={onScroll}>
-              {filterValues.length === 0 ? (
-                <FilterBarEmptyStateContainer>
-                  <EmptyStateSmall
-                    title={t('No filters are currently added')}
-                    image="filter.svg"
-                    description={
-                      canEdit &&
-                      t(
-                        'Click the button above to add a filter to the dashboard',
-                      )
-                    }
-                  />
-                </FilterBarEmptyStateContainer>
-              ) : (
-                <FilterControls
-                  dataMaskSelected={dataMaskSelected}
-                  directPathToChild={directPathToChild}
-                  onFilterSelectionChange={handleFilterSelectionChange}
-                />
-              )}
-            </div>
-          )}
-          <ActionButtons
-            width={width}
-            onApply={handleApply}
-            onClearAll={handleClearAll}
-            dataMaskSelected={dataMaskSelected}
-            dataMaskApplied={dataMaskApplied}
-            isApplyDisabled={isApplyDisabled}
-          />
-        </Bar>
-      </BarWrapper>
-    </FilterBarScrollContext.Provider>
+  const actions = useMemo(
+    () => (
+      <ActionButtons
+        orientation={orientation}
+        width={verticalConfig?.width}
+        onApply={handleApply}
+        onClearAll={handleClearAll}
+        dataMaskSelected={dataMaskSelected}
+        dataMaskApplied={dataMaskApplied}
+        isApplyDisabled={isApplyDisabled}
+      />
+    ),
+    [verticalConfig?.width, dataMaskSelected, dataMaskApplied, isApplyDisabled],
```

Comment:
We have many missing dependencies here. I suggest not using `useMemo` if we can't handle the dependencies because it will be using stale values.